src/quo/components/slideshow/slider_bar/view.cljs

```diff
@@ -0,0 +1,162 @@
+(ns quo.components.slideshow.slider-bar.view
+  (:require
+    [quo.components.slideshow.slider-bar.schema :as component-schema]
+    [quo.components.slideshow.slider-bar.style :as style]
+    [quo.theme]
+    [react-native.core :as rn]
+    [react-native.reanimated :as reanimated]
+    [schema.core :as schema]))
+
+(def ITEM_SIZE 8)
+(def ITEM_SPACING ITEM_SIZE)
+(def MAX_LENGTH 6)
+(def BAR_WIDTH (* (+ ITEM_SPACING ITEM_SIZE) MAX_LENGTH))
+
+(def MICRO_SCALE 0.25)
+(def SMALL_SCALE 0.5)
+(def MEDIUM_SCALE 0.75)
+(def DEFAULT_SCALE 1)
+
+(def ITEMS_BEFORE_SCROLL
+  (/ MAX_LENGTH 2))
+
+(defn- calc-relative-index
+  "Calculates item index in visible area"
+  [index active-index total-length]
+  (let [last-index (dec total-length)]
+    (cond
+      (< active-index ITEMS_BEFORE_SCROLL)   index
+      (> active-index
+         (- last-index ITEMS_BEFORE_SCROLL)) (+ (- index total-length) MAX_LENGTH)
+      :else                                  (+ (- index active-index) ITEMS_BEFORE_SCROLL))))
+
+(defn- first-item
+  "Calculates scale for first item in visible area"
+  [{:keys [active-index]}]
+  (cond
+    (= active-index ITEMS_BEFORE_SCROLL) MEDIUM_SCALE
+    (> active-index ITEMS_BEFORE_SCROLL) SMALL_SCALE
+    :else                                DEFAULT_SCALE))
+
+(defn- second-item
+  "Calculates scale for second item in visible area"
+  [{:keys [active-index]}]
+  (if (> active-index ITEMS_BEFORE_SCROLL)
+    MEDIUM_SCALE
+    DEFAULT_SCALE))
+
+(defn- last-item
+  "Calculates scale for last item in visible area"
+  [{:keys [active-index total-length]}]
+  (let [last-index (dec total-length)]
+    (if (> active-index (- last-index ITEMS_BEFORE_SCROLL))
+      MEDIUM_SCALE
+      SMALL_SCALE)))
+
+(defn- butlast-item
+  "Calculates scale for butlast item in visible area"
+  [{:keys [active-index total-length]}]
+  (if (> active-index (- (dec total-length) ITEMS_BEFORE_SCROLL))
+    DEFAULT_SCALE
+    MEDIUM_SCALE))
+
+(defn- get-scale
+  [index active-index total-length]
+  (let [relative-index (calc-relative-index index active-index total-length)]
+    (if (or (= index active-index)
+            (< total-length MAX_LENGTH))
+      DEFAULT_SCALE
+      (cond
+        (< relative-index 0) MICRO_SCALE
+        (= relative-index 0) (first-item {:active-index active-index})
+        (= relative-index 1) (second-item {:active-index active-index})
+        (= relative-index 4) (butlast-item {:total-length total-length
+                                            :active-index active-index})
+        (= relative-index 5) (last-item {:total-length total-length
```

Comment:
For these functions (`first-item`, etc) we could be more succinct and pass positional args instead of a map.

Reply:
good point!